src/screens/AddSiteScreen.js

```diff
@@ -1,204 +1,339 @@
+    };
+
+    onUpdateSiteDirectionsText = (newDirectionsText) => {
+        this.props.updateSiteDirectionsText({siteDirectionsText: newDirectionsText})
+    };
+
+    onUpdateSiteNearestTownText = (newNearestTownText) => {
+        this.props.updateSiteNearestTownText({siteNearestTownText: newNearestTownText})
+    };
+
+    onUpdateAccessibilityOption = (newAccessibilityOption) => {
+        this.props.updateAccessibilityOption({accessibilityOption: newAccessibilityOption})
+    };
+
+    onUpdateFacilitiesOption = (newFacilitiesOption) => {
+        this.props.updateFacilitiesOption({facilitiesOption: newFacilitiesOption})
+    };
+
+    onClickReset = () => {
+        this.props.resetAddScreenFields();
+    };
+
+    render() {
+        const {buttonStyle, headerTitle, largeTextInput, modalStyle, lastButtonStyle, exitOrResetStyle} = styles;
+        const {latitudeText, longitudeText, siteTitleText, siteDescriptionText, siteDirectionsText, siteNearestTownText, accessibilityOption, facilitiesOption} = this.props;
+
+        return (
+            <View>
+                <Card>
+                    <Text>
+                        {site_description}
+                    </Text>
+                </Card>
+                <Button
+                    onPress={() => this.openModal()}
+                    title="Open modal"
+                    large
+                    rounded={true}
+                    buttonStyle={buttonStyle}
+                    icon={{name: 'plus', type: 'font-awesome'}}
+                    title={add_site}
+                >
+                    {upload}
+                </Button>
+
+                <Modal
+                    visible={this.state.modalVisible}
+                    animationType={'slide'}
+                    onRequestClose={() => this.closeModal()}
+                >
+                    <ScrollView style={modalStyle}>
+                        <View style={exitOrResetStyle}>
+                            <Icon
+                                type='font-awesome'
+                                name='times-circle'
+                                onPress={() => this.closeModal()}
+                            />
+
+                            <Button
+                                title="Reset"
+                                onPress={this.onClickReset}
+                                backgroundColor="rgba(0,0,0,0)"
+                                color="rgba(0,122,255,1)"
+                            />
+
+                        </View>
+
+                        <Text h2
+                              style={headerTitle}
+                        >
+                            {location}
+                        </Text>
+
+                        <FormLabel>{latitude}</FormLabel>
+                        <FormInput
+                            placeholder={latitude_placeholder}
+                            value={latitudeText}
+                            onChangeText={this.onUpdateLatitudeText}
```

Comment:
All of the fields below now follow this pattern of React 'controlled-inputs', where they get their values from props and pass changes out via the actions.